### ISSUE TITLE
OK GUYS I OPENED UP ISSUES GO CRAZY!!!!

### ISSUE THREAD

**Issue by pjreddie** [pjreddie/darknet#72]
![giphy-1](https://user-images.githubusercontent.com/302108/28095734-ff67f906-6658-11e7-9cff-e4498de5d2a5.gif)

**Comment by Grabber** [comment]
@pjreddie, close this issue: https://github.com/pjreddie/darknet/issues/1

**Comment by pjreddie** [comment]
YOUR WISH IS MY COMMAND

![battle](https://user-images.githubusercontent.com/302108/28095909-f042e0e8-6659-11e7-85d8-af0ca2828d6e.gif)

**Comment by Grabber** [comment]
@pjreddie, when are you going to create a real community to move on from an academic project to a industry leader framework?

**Comment by luiscosio** [comment]
@pjreddie Same here, we should create a real community. I'm more than interested in participating with time/resources.

**Comment by Grabber** [comment]
@luiscosio, there are many open challenges.

How to keep Joseph focused on doing what he does best while letting the code base change for community? We are not talking about a complex system like Docker... we are talking about a code base where math concepts is everywhere and a single mistake break everything.

What is your suggestion?

**Comment by luiscosio** [comment]
@Grabber There are several things we could do, besides working on the code base.

From the top of my mind, create a website, wiki (there is a bunch of amazing documentation spread around that we could concentrate), documentation, automatic Docker builds, installation guides, etc.

AlexeyAB (https://github.com/AlexeyAB/darknet) is doing some pretty interesting things in his Windows fork that could be incorporated.

There are several commits on other repos to parse the stdout with Python, could be interesting to attract more people to the community.

Also all the tools to train YOLO for custom classes.

There is a lot of stuff we can do.

**Comment by Grabber** [comment]
@pjreddie, Houston we have a Darknet community or not?

**Comment by WillieMaddox** [comment]
We should also start thinking about adding some unit tests.

**Comment by Grabber** [comment]
@WillieMaddox, I think it is the unique way to guarantee layers don't broke. A microframework to run over the layers with previous known values and checkpoints.

**Comment by loretoparisi** [comment]
💯

**Comment by hot-cheeto** [comment]
my hero ! :O

**Comment by goatShaver** [comment]
I'm interested in contributing and/or participating. 
As someone new to darknet, there were many little things that were difficult for me:

1.) explanation of variables in .cfg file
2.) full difference between v1 and v2 variables and weight files
3.) Python wrapper around the training so that we can iterate various variables and computer things like anchors once the training data has been defined. 

Other things such as visualization tools etc. would be amazingly useful as would some guidelines on training many classes - (There are many tutorials for 1 or 2 class training, however I am trying 20 classes with little success and I keep digging into the code to figure out where things are weird because the results I'm seeing are non-nonsensical. 

Definitely want to help advance things.

**Comment by braca51e** [comment]
I'd also like to add more loss functions and even back-propagate the error for more than one output. Anyways, I would love to contribute should the community be created.

**Comment by stark7** [comment]
with the risk of sounding like the total noob that I am, may I please have a link to where all the hyperparameters are explained? Stuff like:

[net]
batch=1
subdivisions=1
width=416
height=416
channels=3
momentum=0.9
decay=0.0005
angle=0
saturation = 1.5
exposure = 1.5
hue=.1

**Comment by TheMikeyR** [comment]
@stark7 https://github.com/AlexeyAB/darknet/issues/279
